### DISCUSSION TITLE
How can I keep the adorner not being clipped by the adorned control?

### DISCUSSION THREAD

**Discussion by LZD121** [AvaloniaUI/Avalonia#10629] [Q&A] [answered]
I attached a adorner to a control, but everything in the adorner which out of the control's bounds is clipped.
The control is a rectangle , and the attached adorner should display a ellipse at each corner of the rectangle.
As shows below, all the ellipses are cut by half.
![image](https://user-images.githubusercontent.com/59716053/224333384-876ce9bc-0ab7-4389-bdf7-b268ff905df2.png)

This is not what I want.
How can I do to display the whole adorner?

**Comment by maxkatz6** [comment, marked as answer]
`<Control AdornerLayer.IsClipEnabled="false" />` I believe.

**Reply by LZD121** [reply]
Thank you! But I tried before and it didn't work. That's why I come to ask...

**Reply by LZD121** [reply]
My control is pretty simple. Just a rectangle in a ContentControl.
![image](https://user-images.githubusercontent.com/59716053/224335835-15480c0e-72d0-47f0-9f2d-001e52d4bda4.png)

**Reply by timunie** [reply]
which version of Avalonia? There was a fix for it recently if you are on 11.0 preview versions. Check out latest nightly if you're using 11.0 previews

**Reply by LZD121** [reply]
I am using preview5, is that too old?

**Reply by LZD121** [reply]
I have updated to Nightly version and it indeed is fixed, thanks!
But why can't I override Render method in a Canvas now? 
![image](https://user-images.githubusercontent.com/59716053/224346158-4839f89b-ca17-4410-a795-8e49895e6926.png)

Why does it become sealed ??

**Reply by maxkatz6** [reply]
If you want a custom rendered control, inherit a Control and override Render.
See https://github.com/AvaloniaUI/Avalonia/issues/10066

**Reply by LZD121** [reply]
I see, thanks!
